File: services/logger/tests/listener.py
#!/usr/bin/env python

import sys
import qi
import qi.logging

@qi.singleThreaded()
class ListenerCallback():
    def onMessage(self, mess):
        # mess is a dictionnary with all known LogMessage information.
        print mess['message'], "with id", mess['id']

if __name__ == "__main__":
    app = qi.Application()
    app.start()

    logmanager = app.session.service("LogManager")
    listener = logmanager.createListener()
    listener.addFilter("*", qi.logging.DEBUG)
    listener.addFilter("stats.*", qi.logging.SILENT)
    listener.addFilter("core.naoqi", qi.logging.SILENT)
    listener.onLogMessage.connect(ListenerCallback().onMessage)

    app.run()

File: services/logger/tests/listener.cpp
#include <iostream>

#include <boost/thread/mutex.hpp>

#include <qi/applicationsession.hpp>

#include <qicore/logmessage.hpp>
#include <qicore/logmanager.hpp>
#include <qicore/loglistener.hpp>

// More information on logmessage document into LogManager API
static boost::mutex m;
void onMessage(const qi::LogMessage& msg)
{
  boost::mutex::scoped_lock l(m);
  std::stringstream ss;
  ss << msg.category << " " << msg.source << " " << msg.message;
  std::cout << ss.str() << std::endl;
}

int main(int argc, char** argv)
{
  qi::ApplicationSession app(argc, argv);
  app.start();

  app.loadModule("qicore");
  qi::LogManagerPtr logger = app.session()->service("LogManager");
  qi::LogListenerPtr listener = logger->createListener();
  listener->clearFilters();
  listener->setLevel(qi::LogLevel_Info);
  listener->onLogMessage.connect(&onMessage);

  app.run();
}
